src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/azure/AzureProvider.java

```diff
@@ -546,53 +592,100 @@ public void deleteInfrastructure(Infrastructure infrastructure) {
     }
 
     @Override
-    public String addToInstancePublicIp(Infrastructure infrastructure, String instanceId) {
+    public String addToInstancePublicIp(Infrastructure infrastructure, String instanceId, String optionalDesiredIp) {
```

Comment:
Maybe it can be better design to split addToInstancePublicIp() into several submethods: e.g. one to "retrieve the desired public IP address or create a new one", another one to "create a new/secondary network interface", etc.